### ISSUE TITLE
Calls fail on hotspots that only allow web/mail.

### ISSUE THREAD

**Issue by navid-zamani** [signalapp/Signal-Android#11196]
Here in Germany, a popular router is Fritz!Box. It has a hotspot feature that is used by many small businesses (doctor, bicycle shop, etc). That hotspot, by default only allows browsing the web and receiving e-mail and what's necessary to do that. (Like DNS, but only the router's DNS forwarder.)

Unfortunately, this means that Signal calling never connects on many wifi hotspots here in Germany.

But of course it should. Even if abusing port 80 or 443 for something other than HTTP(S) is technically wrong.

I find it surprising, that a security and privacy conscious messenger can be blocked by something that simple. Surely any dictatorship would find it trivial to stop Signal, or at least calls. (I've also read all UDP ports need to be open. … Another thing unlikely in any such situation.)

So the purpose of this bug is to get Signal to work anywhere. Even if just a single port is open and only TCP is allowed. Ideally even in cases where only a whitelist of domains or IP addresses are allowed. (In the long run, I imagine modules for sending messages through various public services and using steganography, e.g. sending data inside images whose encrypted links are posted on Twitter. But that is definitely not part of this bug. :)

I would already be OK with it just working on those Fritz!Box hotspots. (They have great online documentation and support, and are open to concepts like privacy and techie things. So looking up what their firewall lets through, should be trivial. [If necessary, I can do that.])
